apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs

```diff
@@ -101,11 +101,23 @@ public sealed class ReliableStreamReader : IDisposable
                 await ConnectAsync().ConfigureAwait(false);
 
                 Debug.Assert(_currentCall != null, "_currentCall != null");
-                cancellationToken.ThrowIfCancellationRequested();
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    _currentCall?.Dispose();
+                    _currentCall = null;
+                    cancellationToken.ThrowIfCancellationRequested();
+                }
 
                 for (int i = 0; i <= _resumeSkipCount; i++) {
                     //This calls a simple movenext on purpose.  If we get an error here, we'll fail out.
-                    _isReading = await _currentCall.ResponseStream.MoveNext(cancellationToken).WithSessionChecking(() => _session).ConfigureAwait(false);
+                    //TODO(benwu): pass in cancellationtoken and capture cancel to clean up _currentCall if needed.
+                    _isReading = await _currentCall.ResponseStream.MoveNext(CancellationToken.None).WithSessionChecking(() => _session).ConfigureAwait(false);
```

Comment:
An alternative approach would be to use `Task.WhenAny` with the `MoveNext` task and a "just monitor cancellation token" task (I think there's a method to produce one of those). That way we wouldn't have an unbounded delay between token cancellation and response.

This PR is better than the previous code, so I'm happy for it to be merged, but the unbounded nature worries me at the moment.